### PR TITLE
Utf8Literals instead hand-crafted byte-sequences

### PR DESCRIPTION
**Pull request by gfoidl**
Additional:

* remove `movsxd`, as they're not needed
* used C# language to denote if an argument should be ignored

### CODE REVIEW

File: csFastFloat/Constants/Utf8Literals.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace csFastFloat
+{
+  internal partial class Utf8Literals
+  {
+    [StringLiteral.Utf8("infinity")]
```
Comment (gfoidl) [inline csFastFloat/Constants/Utf8Literals.cs:7]: This way it's much nicer 😄 
If you see in Visual Studio errors regarding these definitions, please re-start Visual Studio.
The editor needs to hook up the [source generator](https://devblogs.microsoft.com/dotnet/introducing-c-source-generators/).

File: csFastFloat/FastFloatParser.cs [hunk 2]
```diff
@@ -676,7 +659,7 @@ unsafe static internal ParsedNumberString ParseNumberString(byte* p, byte* pend,
         {
           return answer;
         }
-        if (!Utils.is_integer(*p, out uint digit) && (*p != decimal_separator)) // culture info ?
+        if (!Utils.is_integer(*p, out uint _) && (*p != decimal_separator)) // culture info ?
```
Comment (gfoidl) [inline csFastFloat/FastFloatParser.cs:662]: This is pure stylistic, to denote that the arugment isn't needed.
Emitted machine code will be the same.

Comment (mburbea) [inline csFastFloat/FastFloatParser.cs:662]:
```suggestion
        if (!Utils.is_integer(*p, out _) && (*p != decimal_separator)) // culture info ?
```
It's a nit but this is proper discard syntax.

Comment (gfoidl) [reply csFastFloat/FastFloatParser.cs]: Thanks! Updated e8ddb0f665d9a5521325428c425de288b176e5b4

**Comment by lemire** [comment]
This is now a PR on our main branch, to be merged once people are happy.

**Comment by nietras** [comment]
this PR can be replaced with new `"nan"u8` syntax and similar

**Pull request closed without merging**